src/dev/EventList.tsx

```diff
@@ -1,20 +1,63 @@
-import React, { FC, ReactElement, StrictMode } from 'react';
+import React, { FC, ReactElement, StrictMode, useEffect } from 'react';
 import ReactDOM from 'react-dom';
+import type { Instrument, InstrumentList, Listener } from '@finos/fdc3';
 
 import { Provider } from '@aiera/client-sdk/components/Provider';
 import { useMessageListener } from '@aiera/client-sdk/lib/msg';
 import { Auth } from '@aiera/client-sdk/modules/Auth';
 import { EventList } from '@aiera/client-sdk/modules/EventList';
 import '@aiera/client-sdk/css/styles.css';
 
-const App: FC = (): ReactElement => {
+const useMessageBus = () => {
     const bus = useMessageListener(
         'instrument-selected',
         (msg) => {
-            console.log(`Sending ${JSON.stringify(msg)} to platform`);
+            if (window.fdc3 && msg.data) {
+                const context = {
+                    type: 'fdc3.instrument',
+                    // FDC is really poorly typed, so need to cast this to
+                    // something more generic
+                    id: msg.data as { [key: string]: string },
+                };
+                void window.fdc3.broadcast(context);
+            }
         },
         'out'
     );
+
+    useEffect(() => {
+        const listeners: Listener[] = [];
+        if (window.fdc3) {
+            listeners.push(
+                window.fdc3.addContextListener('fdc3.instrument', (_context) => {
+                    const context = _context as Instrument;
```

Comment:
So you have to cast the `context` from the `fdc3` listener to their own defined type (`Instrument`)?

Reply:
Yea, the typings aren't done all that well. They do have types each each of these, but they didn't split out separate versions of the `addContextListener` for each context type so you have to either use `any` or cast, since both are technically "unsafe" at least this shows the intention more clearly.